src/import/k8s.ts

```diff
@@ -64,6 +65,26 @@ export class ImportKubernetesApi extends ImportBase {
     const typeGenerator = new TypeGenerator({
       definitions: schema.definitions,
       exclude: this.options.exclude,
+      renderTypeName: (def: string) => {
+        const type = def.split('.');
+
+        if (type.length < 2) {
+          // this can happen with synthetic definitions we inject.
+          // e.g KubeMutatingWebhookConfigurationV1Beta1Props
+          return def;
+        }
+
+        const basename = type[type.length - 1];
+        const version = type[type.length - 2];
+
+        if (!VERSION_REGEX.test(version)) {
```

Comment:
@Chriscbr have a look at this? previously its used to apply this to utility classes like `IntOrString` which turned into `IntOrStrintIntstr`

Reply:
This looks OK. But I think we should avoid duplicating this kind of logic by using the `parseApiTypeName()` utility function (it might require some changes, since it currently throws errors if there's no version). I'll approve and let you fix this if you'd like